tests/testthat/test-ddi-ratio.R

```
@@ -33,3 +35,83 @@ test_that("'getSmartZoomLimits' always includes the 2-fold error range for resid
   expect_gt(getSmartZoomLimits(10, residualsVsObserved = TRUE)$max, 2)
   expect_gt(getSmartZoomLimits(1.1, residualsVsObserved = TRUE)$max, 2)
 })
+
+test_that("Guest et al. values are appropriately read", {
+  ddiRatioPlan1 <- list(
+    PKParameters = c("AUC", "CL","CMAX"),
+    GuestDelta = list(
+      list(Value = 1.2, PKParameters = c("AUC", "CL")),
+      list(Value = 1.3, PKParameters = c("CMAX"))
+    )
+  )
+  # Guest value defined but not exported in artifacts
+  ddiRatioPlan2 <- list(
+    PKParameters = c("AUC", "CL"),
+    GuestDelta = list(
+      list(Value = 1.2, PKParameters = c("AUC", "CL")),
+      list(Value = 1.3, PKParameters = c("CMAX"))
+    )
+  )
+  ddiRatioPlan3 <- list(
+    PKParameters = c("AUC", "CL","CMAX"),
+    GuestDelta = list(list(Value = 1.2, PKParameters = c("AUC", "CL")))
+  )
+  ddiRatioPlan4 <- list(
+    PKParameters = c("AUC", "CL","CMAX"),
+    GuestDelta = 1.2
+  )
+  ddiRatioPlan5 <- list(PKParameters = c("AUC", "CL","CMAX"))
+  
+  expect_equal(
+    getGuestDeltaFromConfigurationPlan(ddiRatioPlan1),
+    list(AUC = 1.2, CL = 1.2, CMAX = 1.3)
+  )
+  expect_equal(
+    getGuestDeltaFromConfigurationPlan(ddiRatioPlan2),
+    list(AUC = 1.2, CL = 1.2, CMAX = 1.3)
+  )
```

should throw an error (s. above)